db/write_thread.cc

```diff
@@ -355,7 +358,12 @@ void WriteThread::BeginWriteStall() {
 void WriteThread::EndWriteStall() {
   MutexLock lock(&stall_mu_);
 
+  // Unlink write_stall_dummy_ from the write queue. This will unblock
+  // pending write threads to enqueue themselves
   assert(newest_writer_.load(std::memory_order_relaxed) == &write_stall_dummy_);
+  if (write_stall_dummy_.link_older) {
```

Comment:
can we assert(write_stall_dummy_.link_older != nullptr)? If we are in the write thread, then there must be at least one write in the queue.